control-plane/agents/node/src/server.rs

```diff
@@ -176,6 +182,62 @@ impl ServiceSubscriber for ServiceHandler<GetNodes> {
     }
 }
 
+#[async_trait]
+impl ServiceSubscriber for ServiceHandler<GetBlockDevices> {
+    async fn handler(&self, args: Arguments<'_>) -> Result<(), Error> {
+        let request: ReceivedMessage<GetBlockDevices> =
+            args.request.try_into()?;
+        let store: &NodeStore = args.context.get_state()?;
+        let nodes = store
+            .get_nodes()
+            .await
+            .into_iter()
+            .filter(|n| n.id == request.inner().node)
+            .collect::<Vec<_>>();
+
+        if nodes.is_empty() {
+            return Err(Error::ServiceError {
+                message: format!(
+                    "Node with id {} not found",
+                    request.inner().node
+                ),
+            });
+        }
+
+        // Only expect one node to match the given ID.
+        assert_eq!(nodes.len(), 1);
+
+        let mut client = MayastorClient::connect(format!(
+            "http://{}",
```

Comment:
What if HTTPS is should be used?

Reply:
At the moment all the gRPC connections from the control plane agents to Mayastor use HTTP. See [here](https://github.com/openebs/Mayastor/blob/e2930372fb7241d8b62364d074db171b2273c69f/control-plane/agents/common/src/wrapper/v0/node_traits.rs#L10) in node_traits.rs. This gets used by any service implementing this trait. If we wish to change this, I can open a ticket for it.